.github/workflows/agent-build.yml

```diff
@@ -14,54 +14,84 @@ on: [push]
-      DOCKER_HUB_PASSWORD: ${{ secrets.DOCKER_HUB_PASSWORD_PROD_ACCOUNT }}
-
-  docker-api:
-    uses: scalyr/scalyr-agent-2/.github/workflows/docker-and-k8s.yml@master
-    with:
-      image-type: docker-api
-      publish: false
-    secrets:
-      SCALYR_PROD_CLOUDTECH_TESTING_WRITE_TOKEN: ${{ secrets.SCALYR_PROD_CLOUDTECH_TESTING_WRITE_TOKEN }}
-      SCALYR_CLOUDTECH_TESTING_DEV_SCALYR_READ_API_KEY: ${{ secrets.SCALYR_CLOUDTECH_TESTING_DEV_SCALYR_READ_API_KEY }}
-      DOCKER_HUB_USERNAME: ${{ secrets.DOCKER_HUB_USERNAME_PROD_ACCOUNT }}
-      DOCKER_HUB_PASSWORD: ${{ secrets.DOCKER_HUB_PASSWORD_PROD_ACCOUNT }}
-
-  k8s:
-    uses: scalyr/scalyr-agent-2/.github/workflows/docker-and-k8s.yml@master
-    with:
-      image-type: k8s
-      publish: true
+  test-images:
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        image-type: [ "docker-json", "docker-syslog", "docker-api", "k8s" ]
+        image-distro-name: [ "buster", "alpine" ]
+        python-version: [ "3.8.10" ]
+
+    env:
+      DOCKER_BUILDKIT: 1
+      # Set this variable to tell the agent build code that it runs in CI/CD and it needs to use caching.
+      AGENT_BUILD_IN_CICD: "1"
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Install python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Perform the deployment of the test environment
+        uses: ./.github/actions/perform-deployment
+        with:
+          deployment-name: "test_environment"
+
+      - name: Sanity check deployments logic by running internal tests
+        run: |
+          python3 -m pytest agent_build/tools/tests
+
+      - name: Set up QEMU
+        id: qemu
+        uses: docker/setup-qemu-action@v1
+        with:
+          image: tonistiigi/binfmt:qemu-v6.1.0
+          platforms: all
+
+      - name: Perform the build of the base docker image in the deployment.
+        uses: ./.github/actions/perform-deployment
+        with:
+          deployment-name: ${{ matrix.image-type }}-${{ matrix.image-distro-name }}
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+        with:
+          driver-opts: network=host
+
+      - name: Start minikube for the test of the kubernetes build
+        if: matrix.image-type == 'k8s'
+        shell: bash
+        run: |
+          minikube start
+
+      - name: Run image test
+        env:
+          SCALYR_API_KEY: ${{ secrets.SCALYR_PROD_CLOUDTECH_TESTING_WRITE_TOKEN }}
+        run: |
+
+          python3 tests/package_tests/run_package_test.py package-test \
+           "${{ matrix.image-type }}-${{ matrix.image-distro-name }}_test" \
+            --name-suffix "_${{ github.run_number }}${{ github.run_attempt }}" \
+
+
+  publish-images:
+    if: ${{ github.ref == 'refs/heads/improve-caching' || github.ref_type == 'tag' }}
```

Comment:
I assume you will change this to master branch name after it's confirmed working, right?

Reply:
right